### PR TITLE
Feature/stp drive with ball tactic

### LINKED ISSUE #860: DriveWithBall
At this point the ball is already with the robot.
The robot will turn with the ball until it can follow a straight line from the path planning
Then it will drive in a straight line with the ball. 
This process repeats until the robot gets to the desired position

Failure: loseBall 
Reset: Angle of robot != angle of (Ball - Robot)
ie robot is no longer facing the ball

### PR DESCRIPTION
**Pull request by timovdk**
Rotates towards the ball and drives to the target position with the ball. The limiting during driving and dribbling is hardcoded as of now, depends on test findings on how to implement this better. 
Closes #860

### CODE REVIEW

File: src/stp/new_plays/TestPlay.cpp [hunk 1]
```diff
@@ -49,8 +49,7 @@ void TestPlay::assignRoles() noexcept {
             stpInfos.emplace(roleName, StpInfo{});
             stpInfos[roleName].setRobot(robot);
             stpInfos[roleName].setField(*world->getField());
-
-            // TODO calculate additional info
+            stpInfos[roleName].setBall(world->getWorld()->getBall());
```
Comment (timovdk) [inline src/stp/new_plays/TestPlay.cpp]: This is already done in the base class, woops
```suggestion
```

File: src/stp/new_tactics/GetBall.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ namespace rtt::ai::stp::tactic {
         // the robot will go to the position of the ball
         double ballDistance = (ballPosition - robotPosition).length();
         Vector2 newRobotPosition = robotPosition + (ballPosition - robotPosition).stretchToLength(ballDistance - Constants::ROBOT_RADIUS());
-        if (ballDistance < 3 * Constants::ROBOT_RADIUS()){
+        if (ballDistance < Constants::ROBOT_RADIUS()){
```
Comment (timovdk) [inline src/stp/new_tactics/GetBall.cpp]: Woops, thought this was my file lol
```suggestion
        if (ballDistance < 3 * Constants::ROBOT_RADIUS()){
```

File: src/stp/Skill.cpp [hunk 3]
```diff
@@ -53,16 +53,20 @@ void Skill::refreshRobotCommand() noexcept {
...
 
     if (std::isnan(limitedVel.x) || std::isnan(limitedVel.y)) {
         RTT_ERROR("Robot will have NAN: " + std::string{name()} + "!\nrobot: " + std::to_string(robot.value()->getId()));
     }
+
+    // TODO: When using the dribbler, we probably want to limit the vel a bit smarter than this...
+    if(this->command.dribbler() != 0) {
+        limitedVel.x /= 3;
+        limitedVel.y /= 3;
+    }
```
Comment (RobotJesse) [inline src/stp/Skill.cpp]: I think this will make the speed go to 0

**Review by RobotJente — APPROVED** [review]

File: src/stp/new_plays/TestPlay.cpp [hunk 4]
```diff
@@ -48,6 +48,8 @@ void TestPlay::assignRoles() noexcept {
 
             stpInfos.emplace(roleName, StpInfo{});
             stpInfos[roleName].setRobot(robot);
+            stpInfos[roleName].setField(*world->getField());
+
```
Comment (RobotJente) [inline src/stp/new_plays/TestPlay.cpp]:
```suggestion
```

**Pull request merged**